modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/FlagHighConfidenceOrthologs.pm

```diff
@@ -55,112 +59,138 @@ sub fetch_input {
-    my $sql_reset = "UPDATE homology JOIN homology_member USING (homology_id) SET is_high_confidence = 0 WHERE $homology_filter AND NOT ($high_confidence_condition)";
-    $self->compara_dba->dbc->do($sql_sethc, undef, $mlss_id);
-    $self->compara_dba->dbc->do($sql_reset, undef, $mlss_id);
-
-    # Get some statistics for the mlss_tag table
-    my $sql_hc_count         = "SELECT SUM(is_high_confidence) FROM homology WHERE $homology_filter";
-    my $sql_hc_per_gdb_count = "SELECT genome_db_id, COUNT(DISTINCT gene_member_id) FROM homology JOIN homology_member USING (homology_id) JOIN gene_member USING (gene_member_id) WHERE $homology_filter AND is_high_confidence = 1 GROUP BY genome_db_id";
-    my ($n_hc) = $self->compara_dba->dbc->db_handle->selectrow_array($sql_hc_count, undef, $mlss_id);
-    my $hc_per_gdb = $self->compara_dba->dbc->db_handle->selectall_arrayref($sql_hc_per_gdb_count, undef, $mlss_id);
-    
+    my $mlss                = $self->param('mlss');
+    my $range_label         = $self->param('range_label');
+    my $range_filter        = $self->param('range_filter')->{$range_label};
+    my $conditions          = $self->param('conditions');
+    my $external_conditions = $self->param('external_conditions');
+    my $n_hom               = $self->param('num_homologies'),
+
+    my $homology_file = $self->param_required('homology_file');
+    my $wga_file      = $self->param('wga_file');
+    my $goc_file      = $self->param('goc_file');
+    my $output_file   = $self->param_required('high_conf_file');
+    $self->run_command( "mkdir -p " . dirname($output_file)) unless -e dirname($output_file);
+
+    my ( $wga_coverage, $goc_scores );
+    $wga_coverage = $self->_parse_flatfile_into_hash($wga_file, $range_filter) if $wga_file;
+    $goc_scores   = $self->_parse_flatfile_into_hash($goc_file, $range_filter) if $goc_file;
+
+    open(my $hfh, '<', $homology_file) or die "Cannot open $homology_file for reading";
+    open(my $ofh, '>', $output_file  ) or die "Cannot open $output_file for writing";
+    print $ofh "homology_id\tis_high_confidence\n";
+    my $header_line = <$hfh>;
+    my @header_cols = split( /\s+/, $header_line );
+    my (%hc_counts, $n_hc);
+    while ( my $line = <$hfh> ) {
+        my $row = map_row_to_header($line, \@header_cols);
+        my ($homology_id, $is_tree_compliant, $gdb_id_1, $gm_id_1, $perc_id_1, $gdb_id_2, $gm_id_2, $perc_id_2) = (
+            $row->{homology_id}, $row->{is_tree_compliant}, $row->{genome_db_id}, $row->{gene_member_id}, $row->{identity},
+            $row->{hom_genome_db_id}, $row->{hom_gene_member_id}, $row->{hom_identity},
+        );
+
+        if ( $range_filter ) {
+            next unless $self->_match_range_filter($homology_id, $range_filter);
+        }
+
+        # decide if homology is high confidence
+        my $is_high_conf = ($perc_id_1 >= $conditions->{perc_id} && $perc_id_2 >= $conditions->{perc_id});
+
+        if ( $external_conditions ) {
```

Comment:
Being a reference, `$external_conditions` is always "true". You need to dereference it to check the emptyness